build/ublue-os-just/40-nvidia.just

```diff
@@ -59,15 +59,44 @@ configure-nvidia ACTION="prompt":
     fi
 
 # Switch between Nvidia image and NVK
-toggle-nvk:
+toggle-nvidia ACTION="prompt":
     #!/usr/bin/bash
     CURRENT_IMAGE=$(rpm-ostree status -b --json | jq -r '.deployments[0]."container-image-reference"' | sed -E 's/^.+\/(.+:.+)$/\1/')
     CURRENT_URI=$(rpm-ostree status -b --json | jq -r '.deployments[0]."container-image-reference"' | sed -E 's/^(.+\/).+:.+$/\1/')
     NEW_IMAGE=$CURRENT_IMAGE
     if [[ "$CURRENT_IMAGE" =~ "nvidia" ]]; then
-      NEW_IMAGE=${CURRENT_IMAGE/-nvidia/}
+      IS_CURRENTLY_NVIDIA=true
+      echo "Current nvidia driver: Nvidia (From Nvidia Inc, Proprietary)"
     else
-      NEW_IMAGE=${CURRENT_IMAGE/:/-nvidia:}
+      IS_CURRENTLY_NVIDIA=false
+      echo "Current nvidia driver: NVK (Open Source, MESA)"
+    fi
+    OPTION={{ ACTION }}
+    if [ "$OPTION" == "prompt" ]; then
+      echo "${bold}Enable or Disable Nvidia driver${normal}"
+      echo 'What would you like to do?'
+      OPTION=$(ugum choose "Enable" "Disable")
+    elif [ "$OPTION" == "help" ]; then
+      echo "Usage: ujust toggle-nvidia <option>"
+      echo "  <option>: Specify the quick option - 'enable' or 'disable'"
+      echo "  Use 'enable' to Enable the proprietary Nvidia driver from Nvidia Inc (reboot required)."
+      echo "  Use 'disable' to Enable the open source Nvidia driver (reboot required)."
+      exit 0
+    fi
+    if [ "$OPTION" == "Enable" ] || [ "${OPTION,,}" == "enable" ]; then
+      if $IS_CURRENTLY_NVIDIA; then
+        echo "Proprietary nvidia drivers are already enabled."
+        exit 0
+      else
+        NEW_IMAGE=${CURRENT_IMAGE/-main/-nvidia}
```

Comment:
this will not work on bluefin and bazzite since we do not have `-main` images there and future downstreams might not have it either

Reply:
@HikariKnight true, I'm open to suggestions here.

Reply:
@qoijjj dont know a good way to do this other than split the `toggle-nvidia` part into i guess `41-nvk.just` or something and have downstream replace/adjust that file as needed